### ISSUE TITLE
update-handle-prefix runs a very resource intensive (and possibly 'dangerous' discovery clear/recreation) 

### ISSUE THREAD

**Issue by jonas-atmire** [DSpace/DSpace#9805]
## Describe the bug
When running the update-handle-prefix for a repository, it only updates the relevant DSpace objects with the provided prefix, but it still goes through the entire discovery reindexing, which is very resource intensive, and can cause problems

For example:
If there is a repo with a vast amount of items, and there are a small amount of items with an incorrect handle, that would update those items.

The actual discovery reindexing being done even **clears** the entire core.
If the process at that time does not have enough memory, it could be that the discovery indexing fails, resulting in items not existing in discovery anymore, and a separate discovery indexing needing to be ran.

```
 dspace/bin/dspace update-handle-prefix 123456789 987654321

Getting information about handles from database...
In your repository will be updated 6 handles to new prefix 987654321 from original 123456789!

Servlet container (e.g. Apache Tomcat, Jetty, Caucho Resin) must be running.
If it is necessary, please make a backup of the database.
Are you ready to continue? [y/n]: y

Updating handle table... 6 items updated
Updating metadatavalues table... 1 metadata value updated
Handles successfully updated in database.

Re-creating browse and search indexes...
The script has started
(Re)building index from scratch.
```

## To Reproduce
1. Get a random repo with any number of items
2. Add a single item with a different handle
3. Run the update-handle-prefix script
4. While the script goes through the indexing, check discovery -> It will be cleared

## Expected behavior
Only the updated items need to be indexed, nothing else

**Comment by tdonohue** [comment]
I agree that it'd be better to update *only* the items that need it in the index.  However, the current process is not *dangerous*, as it's just running the usual `./dspace index-discovery -b` command (which is also often referenced in upgrade processes, etc).  

That command is a full reindex though, so it can be resource intensive and take a long time for large sites.  So, I do agree that, ideally, the script would be changed to only reindex items that are updated.

Needs a volunteer.
